devito/passes/clusters/factorization.py

```diff
@@ -43,6 +43,71 @@ def factorize(cluster, *args):
     return cluster.rebuild(processed)
 
 
+def collect_special(expr):
```

Comment:
not sure because it's not as simple as calling them back to back... if we go straightly back-to-back there's the risk we unpick something we had done in a previous pass

Does it make sense?

So, I mean, the ordering is important